File: android/app/src/main/kotlin/com/example/hackathon_app/NativeCameraViewFactory.kt
package com.example.hackathon_app

import android.content.Context
import android.view.View
import android.widget.FrameLayout
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory

class NativeCameraViewFactory(
    private val handler: NativeCameraHandler
) : PlatformViewFactory(StandardMessageCodec.INSTANCE) {

    override fun create(context: Context, viewId: Int, args: Any?): PlatformView {
        return NativeCameraView(context, handler)
    }
}

class NativeCameraView(
    context: Context,
    private val handler: NativeCameraHandler
) : PlatformView {

    private val containerView: FrameLayout = FrameLayout(context).apply {
        setBackgroundColor(android.graphics.Color.BLACK)
    }

    init {
        // カメラプレビューのコンテナとして使用
        handler.setCameraPreviewContainer(containerView)
    }

    override fun getView(): View {
        return containerView
    }

    override fun dispose() {
        // Clean up
    }
}

File: ios/Runner/NativeCameraViewFactory.swift
import Flutter
import UIKit

class NativeCameraViewFactory: NSObject, FlutterPlatformViewFactory {
    private var handler: NativeCameraHandler

    init(handler: NativeCameraHandler) {
        self.handler = handler
        super.init()
    }

    func create(
        withFrame frame: CGRect,
        viewIdentifier viewId: Int64,
        arguments args: Any?
    ) -> FlutterPlatformView {
        return NativeCameraView(
            frame: frame,
            viewIdentifier: viewId,
            arguments: args,
            handler: handler
        )
    }

    func createArgsCodec() -> FlutterMessageCodec & NSObjectProtocol {
        return FlutterStandardMessageCodec.sharedInstance()
    }
}

class NativeCameraView: NSObject, FlutterPlatformView {
    private var _view: UIView
    private var handler: NativeCameraHandler

    init(
        frame: CGRect,
        viewIdentifier viewId: Int64,
        arguments args: Any?,
        handler: NativeCameraHandler
    ) {
        _view = UIView(frame: frame)
        _view.backgroundColor = .black
        self.handler = handler
        super.init()

        // カメラプレビューのビューとして使用
        handler.setCameraPreviewContainer(_view)
    }

    func view() -> UIView {
        return _view
    }
}
